EIPS/eip-5173.md

```diff
@@ -0,0 +1,374 @@
+
+### Distribution of NFT Owners’ Future Rewards (FRs)
+
+#### Future Rewards calculation
+
+Any realized profits (P) when an NFT is sold are distributed among the buyers/owners. The previous owners will take a fixed portion of the profix (P), and this portion is called Future Rewards (FRs). The seller takes the rest of the profits.
+
+We define a sliding window mechanism to decide which previous owners will be involved in the profit distribution. Let's imagine the owners as a queue starting from the first hand owner to the current owner. The profit distribution window starts from the previous owner immediately to the current owner and extends towards the first owner, and the size of the windows is fixed. Only previous owners located inside the window will join the profit distribution.  
+
+![Future Rewards calculation formula](../assets/eip-5173/nFR_distribution_formula.png?raw=true)
+
+In this equation:
+
+- P is the total profit, the difference between the selling price minus the buying price;
+- r is buyer reward ratio of the total P;
+- g is the common ratio of successive in the geometric sequence;
+- n is the actual number of owners eligible and participating in the future rewards sharing. To calculate n, we have n = min(m, w), where m is the current number of owners for a token, and w is the window size of the profit distribution sliding window algorithm
+
+#### Converting into Code
+
+```solidity
+
+pragma solidity ^0.8.0;
+//...
+
+/* Assumes usage of a Fixed Point Arithmetic library (prb-math) for both int256 and uint256, and OpenZeppelin Math utils for Math.min. */
+function _calculateFR(uint256 P, uint256 r, uint256 g, uint256 m, uint256 w) pure internal virtual returns(uint256[] memory) {
+        uint256 n = Math.min(m, w);
+        uint256[] memory FR = new uint256[](n);
+
+        for (uint256 i = 1; i < n + 1; i++) {
+            uint256 pi = 0;
+
+            if (successiveRatio != 1e18) {
+                int256 v1 = 1e18 - int256(g).powu(n);
+                int256 v2 = int256(g).powu(i - 1);
+                int256 v3 = int256(P).mul(int256(r));
+                int256 v4 = v3.mul(1e18 - int256(g));
+                pi = uint256(v4 * v2 / v1);
+            } else {
+                pi = P.mul(r).div(n);
+            }
+
+            FR[i - 1] = pi;
+        }
+
+        return FR;
+}
+
+```
+The complete implementation code can be found [here](https://github.com/unTrading/eip-5173-reference-implementation).
+
+## Security Considerations
+
+### Payment Attacks
+
+As this EIP introduces royalty and realized profit rewards collection, distribution, and payouts to the [EIP-721](./eip-721.md) standard, the attack vectors increase. As discussed in EIP-4910 and other mitigations to phishing attacks by Andreas Freund, we recommend reentrancy protection for all payment functions to reduce the most significant attack vectors for payments and payouts.
```

Comment:
Probably shouldn't be referencing EIPs that haven't been merged into the repo yet.

Reply:
Thank you for your review and instructions.

Reply:
Updated.